test/bigtest/tests/coverage-tests.js

```diff
@@ -99,6 +122,58 @@ describe('Coverage tests', () => {
     it('correctly renders the volume', () => {
       expect(monographInteractor.volume).to.have.string(monograph.pti.titleInstance.monographVolume);
     });
+
```

Comment:
Again, this is a complicated component with a bunch of if-case switching that's tough to follow. So [from what I see here](https://github.com/folio-org/ui-agreements/pull/487/files#diff-a21189892d9c449c206496ab1e32c75cR106), we've got six different paths. And each of those paths can either have coverage (of some form), or not have it. So at a minimum, I see a value for 12 different test cases.

And then separate from that, I'd have another unit test suite (`embargo-tests.js`) that just mounts the `Embargo` component and runs through all of it's permutations.

I _frequently_ think exhaustive testing is dumb and a waste of time, but this is definitely not one of those times. I see Coverage and Embargo as very risky components and worth having really bulletproof tests for, especially given how hard it is to experimentally validate everything on a live system.

Reply:
I extended the existing coverage test with the embargo data. So if you recommend a separate embargo-test I would start with that, but leave the existing coverage test untouched as this issue is about embargo and not coverage.

Reply:
So there are two prongs:
1. Tests are needed for Coverage for the six different paths because they change what's fed into Embargo. We cant test those paths via just Embargo.
2. Tests are needed for Embargo to confirm all its various permutations of embargoes. And yeah, those should be in a separate test that just mounts the Embargo component.

Reply:
Just for clarification:
> 1. Tests are needed for Coverage for the six different paths because they change what's fed into Embargo. We cant test those paths via just Embargo.

I kept the existing 6 coverage tests (pci, eresource, line for monograph and serial each) and added only the embargo data. No changes on coverage data.

> 2. Tests are needed for Embargo to confirm all its various permutations of embargoes. And yeah, those should be in a separate test that just mounts the Embargo component.

I added an embargo test for 4 different embargo cases: embargo with Start and End, no embargo, embargo w/o end, embargo w/o start (although this should never occur)

Is that what you expected?